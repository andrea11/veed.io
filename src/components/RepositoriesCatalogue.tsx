import RepositoryCard from "@/components/RepositoryCard";
import {Endpoints} from "@octokit/types";

type RepositoriesCatalogueProps = {
    repositories: Endpoints["GET /search/repositories"]["response"]["data"]["items"]
    title: string

    description?: string
}
export default function RepositoriesCatalogue({repositories, title, description}: RepositoriesCatalogueProps) {
    return (
        <div>
            <div className="mx-auto max-w-2xl py-16 px-4 sm:py-24 sm:px-6 lg:max-w-7xl lg:px-8">
                <h2 className="text-2xl font-bold tracking-tight text-gray-900">{title}</h2>
                {description && <h3>{description}</h3>}

                <div className="mt-6 grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
                    {repositories.map((repository) => (<RepositoryCard repository={repository} key={repository.id}/>))}
                </div>
            </div>
        </div>
    )
}