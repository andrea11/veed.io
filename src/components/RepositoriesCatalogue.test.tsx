import RepositoriesCatalogue from "@/components/RepositoriesCatalogue";
import {render} from "@testing-library/react";

jest.mock("minidenticons", () => ({
    identicon: jest.fn()
}))

describe("RepositoriesCatalogue", () => {
    it("renders correctly", () => {
        const { container } = render(<RepositoriesCatalogue repositories={[]} title={"title"}/>)
        expect(container).toMatchSnapshot()
    })
})