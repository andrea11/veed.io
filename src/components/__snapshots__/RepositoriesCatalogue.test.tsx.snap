// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesCatalogue renders correctly 1`] = `
<div>
  <div>
    <h2
      class="text-2xl font-bold tracking-tight text-gray-900"
    >
      title
    </h2>
    <div
      class="mt-6 grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8"
    />
  </div>
</div>
`;
